integrators/methods.hpp

```diff
@@ -79,19 +84,20 @@ struct SymplecticPartitionedRungeKutta : not_constructible {
 // (the possible composition methods are constrained by the properties of the
 // SPRK).  This struct effects that transformation.
 template<typename SymplecticPartitionedRungeKuttaMethod,
-         SymplecticRungeKuttaNyström::CompositionMethod composition_>
+         serialization::FixedStepSizeIntegrator::CompositionMethod composition_>
 struct AsSymplecticRungeKuttaNyström {
-  static_assert(std::is_base_of<methods::SymplecticPartitionedRungeKutta,
+  static_assert(std::is_base_of<SymplecticPartitionedRungeKutta,
                                 SymplecticPartitionedRungeKuttaMethod>::value,
                 "Method must be derived from SymplecticPartitionedRungeKutta");
-  static_assert(composition_ != SymplecticRungeKuttaNyström::ABA,
+  static_assert(composition_ != serialization::FixedStepSizeIntegrator::ABA,
                 "ABA not supported until C++17");
-  static_assert(SymplecticPartitionedRungeKuttaMethod::first_same_as_last
-                    ? composition_ == SymplecticRungeKuttaNyström::ABA ||
-                          composition_ == SymplecticRungeKuttaNyström::BAB
-                    : composition_ == SymplecticRungeKuttaNyström::BA,
-                "requested |composition| inconsistent with the properties of "
-                "this method");
+  static_assert(
+      SymplecticPartitionedRungeKuttaMethod::first_same_as_last
+          ? composition_ == serialization::FixedStepSizeIntegrator::ABA ||
+                composition_ == serialization::FixedStepSizeIntegrator::BAB
+          : composition_ == serialization::FixedStepSizeIntegrator::BA,
+      "requested |composition| inconsistent with the properties of "
+      "this method");
```

Comment:
Put the "unsupported" `static_assert` after the correctness assertions.
Also consider splitting this static assert into two, so that the message is more explicit (in C++17 the split could be made more legible with `if constexpr`, but in the meantime logic works).
```C++
  static_assert(
      !SymplecticPartitionedRungeKuttaMethod::first_same_as_last ||
      composition_ == serialization::FixedStepSizeIntegrator::ABA ||
      composition_ == serialization::FixedStepSizeIntegrator::BAB
      "requested |composition| must be ABA or BAB for this first-same-as-last method");
  static_assert(
      SymplecticPartitionedRungeKuttaMethod::first_same_as_last ||
      composition_ == serialization::FixedStepSizeIntegrator::BA
      "requested |composition| must be BA for this method which is not first-same-as-last");
  static_assert(composition_ != serialization::FixedStepSizeIntegrator::ABA,
                "ABA not supported until C++17");
```

Reply:
Done.